From 6ab0e059e67fc7b3e040dd1a1bcf7b2c8f8e01e1 Mon Sep 17 00:00:00 2001
From: "Liu, Chuansheng" <chuansheng.liu@intel.com>
Date: Thu, 26 Dec 2013 11:21:55 +0800
Subject: [PATCH 1/1] power_supply: Removing the redundant calling
 power_supply_remove_triggers()

BZ: 161223

Removing the redundant calling power_supply_remove_triggers(), which may cause
memory corruption due to freeing freed memory.

Change-Id: I60f72e293ca9cc32235a360fb82d7e8b6599edad
Signed-off-by: Liu, Chuansheng <chuansheng.liu@intel.com>
---
 drivers/power/power_supply_core.c |    1 -
 1 file changed, 1 deletion(-)

diff --git a/drivers/power/power_supply_core.c b/drivers/power/power_supply_core.c
index 8a12de7..cb13224 100644
--- a/drivers/power/power_supply_core.c
+++ b/drivers/power/power_supply_core.c
@@ -696,7 +696,6 @@ void power_supply_unregister(struct power_supply *psy)
 {
 	cancel_work_sync(&psy->changed_work);
 	sysfs_remove_link(&psy->dev->kobj, "powers");
-	power_supply_remove_triggers(psy);
 	if (IS_CHARGER(psy))
 		power_supply_unregister_charger(psy);
 	power_supply_remove_triggers(psy);
-- 
1.7.10.4

